src/Xrefcheck/Verify.hs

```diff
@@ -408,18 +410,17 @@ verifyReference
 
     checkRef mAnchor referredFile = verifying $ do
       checkReferredFileExists referredFile
-      case M.lookup referredFile repoInfo of
+      case join $ M.lookup referredFile repoInfo of
         Nothing -> pass  -- no support for such file, can do nothing
         Just referredFileInfo -> whenJust mAnchor $
           checkAnchor referredFile (_fiAnchors referredFileInfo)
 
     checkReferredFileExists file = do
-      let fileExists = readingSystem $ doesFileExist file
-      let dirExists = readingSystem $ doesDirectoryExist file
+      let fileExists =
+            any (equalFilePath $ normaliseWithNoTrailing file) $ M.keys repoInfo
```

Comment:
This feels off...

In this line, we're using `Map.keys + any + equalFilePath + normaliseWithNoTrailing` to check if a filepath exists in the map. This implies that the `file` variable might not look exactly like the filepath in the `Map` (e.g. one might contain a leading `./`, the other might not), so we need to use `equalFilePath` to correct that behavior.

However, in line 413, we're using `M.lookup`, which implies that `file` variable _does_ look exactly like the filepaths in the `Map`.

Can you please rebase this on master? The tests in #189 have been merged to master, I have a suspicion some of them might fail.

Reply:
Oh, indeed tests for local references fail. I will fix it, thanks!